packages/bot/src/stores/flyout-store.js

```diff
@@ -160,9 +160,10 @@ export default class FlyoutStore {
         const toolbox         = Blockly.derivWorkspace.toolbox_; // eslint-disable-line
         const is_flyout_click = event.path.some(el => el.classList && el.classList.contains('flyout'));
         const isToolboxClick  = () => toolbox.HtmlDiv.contains(event.target);
+        const isSearchInput   = document.querySelector('input[name="search"]') === event.target;
 
-        if (!is_flyout_click && !isToolboxClick()) {
-            toolbox.clearSelection();
+        if (!is_flyout_click && !isToolboxClick() && !isSearchInput) {
+            this.setVisibility(false);
```

Comment:
We need to keep `toolbox.clearSelection();` as it clears the selected category internally, `clearSelection` calls `setSelectedItem(null)` which in turn sets the visibility of the flyout to `false` by calling `this.setVisibility(false)`.

```suggestion
            toolbox.clearSelection();
```